From: Ubuntu Developers <ubuntu-devel-discuss@lists.ubuntu.com>
Date: Mon, 16 Mar 2026 10:31:00 +0800
Subject: bash53-004

BASH PATCH REPORT
			     =================

Bash-Release:	5.3
Patch-ID:	bash53-004

Bug-Reported-by:	Emanuele Torre <torreemanuele6@gmail.com>
Bug-Reference-ID:	<aHKS0uA8Sf_vMT25@ntoo>
Bug-Reference-URL:	https://lists.gnu.org/archive/html/bug-bash/2025-07/msg00031.html

Bug-Description:

The Linux kernel reports incorrect sizes for files in /sys/block/*/uevent,
leading bash to report a read error when the byte count does not agree
with the file size from fstat(2).
---
 builtins/evalfile.c | 2 ++
 patchlevel.h        | 2 +-
 2 files changed, 3 insertions(+), 1 deletion(-)

diff --git a/builtins/evalfile.c b/builtins/evalfile.c
index 1fee988..3026c5f 100644
--- a/builtins/evalfile.c
+++ b/builtins/evalfile.c
@@ -160,8 +160,10 @@ file_error_and_exit:
       nr = read (fd, string, file_size);
       if (nr >= 0)
 	string[nr] = '\0';
+#if 0
       if (nr != file_size)
 	nr = -1;		/* XXX - didn't get the whole file */
+#endif
     }
   else
     nr = zmapfd (fd, &string, 0);
diff --git a/patchlevel.h b/patchlevel.h
index a1609dc..fa8d6cb 100644
--- a/patchlevel.h
+++ b/patchlevel.h
@@ -25,6 +25,6 @@
    regexp `^#define[ 	]*PATCHLEVEL', since that's what support/mkversion.sh
    looks for to find the patch level (for the sccs version string). */
 
-#define PATCHLEVEL 3
+#define PATCHLEVEL 4
 
 #endif /* _PATCHLEVEL_H_ */
